knowledge-base/how-to-validate-report-parameters-on-the-client.md

```diff
@@ -0,0 +1,145 @@
+description: How to validate report parameters on the client-side
+type: how-to
+page_title: Client-side report parameters validation in HTML5 Report Viewer 
+slug: how-to-validate-report-parameters-on-the-client
+position: 
+tags: viewer,report,reporting, report parameters, validation, client-side validation, report parameters validation
+ticketid:
+res_type: kb
+---
+
+## Environment
+
+<table>
+	<tr>
+		<td>Product</td>
+		<td>Progress® Telerik® Reporting</td>
+	</tr>
+	<tr>
+		<td>Report Viewer</td>
+		<td>Web Viewers</td>
+	</tr>
+</table>
+
+
+## Description
+
+Client-side validation on a report parameter based on the value of another report parameter
+
+## Solution
+
+The following guide shows how to validate a report parameter in [HTML5 ASP.NET MVC Report Viewer](../html5-report-viewer). The approach could be applied to any other Web Report Viewer.
+
+1. Start by creating custom parameter editors by defining them through the [ParameterEditors method](https://docs.telerik.com/reporting/m-telerik-reportviewer-mvc-ireportviewerbuilder-parametereditors) when creating the report viewer widget:
+
+    ```
+    @(Html.TelerikReporting().ReportViewer()
+        .Id("reportViewer1")
+        .ServiceUrl(Url.Content("~/api/reports"))
+        .ReportSource(new UriReportSource() { Uri = "TestReport.trdp" })
+        .ViewMode(ViewMode.Interactive)
+        .ScaleMode(ScaleMode.Specific)
+        .Scale(1.0)
+        .ParameterEditors(
+                  editors => editors
+                      .SingleSelectEditor("createSingleSelectEditor")
+                      .NumberEditor("createNumberEditor")
+            )
+    )
+    ```
+  
+    *Note:* You’ll need to create a [custom editor](https://docs.telerik.com/reporting/html5-mvc-report-viewer-customizing-custom-parameters-editor) for each parameter type that will be validating.
+  
+2. In this example, we have a single select editor with available values and a number parameter that has to be validated based on the first parameter value. That means that we are going to need a combobox editor and simple input editor.
+In this case, we have both UI elements that we are needing to represent our editors. The first one will be a [Kendo ComboBox widget](https://docs.telerik.com/kendo-ui/api/javascript/ui/combobox), which we have from the [Telerik Reporting](https://docs.telerik.com/reporting/html5-report-viewer-widget-requirements) scripts and
+simple [HTMl5 text input](https://www.w3schools.com/tags/tag_input.asp). If you wish to use some Kendo widget, that is not included in a subset of the used by default Kendo UI widgets from Telerik Reporting, you need to add manually the scripts for the additional widget or add [kendo.all.js](https://docs.telerik.com/kendo-ui/intro/installation/cdn-service):
+
+3. Create the editors and applay the validation on [input change](https://api.jquery.com/change/):
+
+    ```
+    var basedParameterName = "Parent", //The parameter name that will be used for validaion
```

Comment:
The naming of "based" and "validating" could be improved. Base: master, parent. Validating: child, dependent.